### PR TITLE
fix: sonar issues

### PR DESCRIPTION
**Pull request by Andrew-Orlov**
## Description

Fix Somar issues

### Jira-link:

https://virtocommerce.atlassian.net/browse/VCST-1897

### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-2.7.0-pr-1346-aaf9-aaf91eee.zip

### CODE REVIEW

File: client-app/shared/catalog/components/product-card-grid.vue [hunk 1]
```diff
@@ -100,7 +100,7 @@
         <template #trigger>
           <router-link
             :to="link"
-            :target="target"
+            :target="browserTarget"
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/product-card-grid.vue:103]: why we can't use $cfg.details_browser_target here?

File: client-app/ui-kit/components/molecules/line-item/vc-line-item.vue [hunk 2]
```diff
@@ -42,7 +42,7 @@
           :disabled="disabled || deleted"
           :to="route"
           :title="name"
-          :target="target"
+          :target="$cfg.details_browser_target"
```
Comment (goldenmaya) [inline client-app/ui-kit/components/molecules/line-item/vc-line-item.vue:45]: bad idea to use $cfg.details_browser_target in ui-kit

File: client-app/ui-kit/components/molecules/variations-button/vc-variations-button.vue [hunk 3]
```diff
@@ -23,26 +23,35 @@
...
+defineEmits<IEmits>();
+
 const props = withDefaults(defineProps<IProps>(), {
   link: "",
   variationsCount: 0,
 });
 
+interface IProps {
+  link?: string;
+  target?: BrowserTargetType;
```
Comment (goldenmaya) [inline client-app/ui-kit/components/molecules/variations-button/vc-variations-button.vue:44]: I think it's better to use local types here.

Comment (Andrew-Orlov) [reply client-app/ui-kit/components/molecules/variations-button/vc-variations-button.vue:44]: We have many cases when ui-kit uses core types.

File: client-app/shared/catalog/components/product-card-grid.vue (hunk 1 above)
Comment (Andrew-Orlov) [reply client-app/shared/catalog/components/product-card-grid.vue:103]: To keep only one logic entry

File: client-app/ui-kit/components/molecules/variations-button/vc-variations-button.vue [hunk 4]
```diff
@@ -23,26 +23,35 @@
 
 <script setup lang="ts">
 import { useBreakpoints } from "@vueuse/core";
+import { toRef } from "vue";
 import { useThemeContext } from "@/core/composables";
 import { BREAKPOINTS } from "@/core/constants";
-import { getLinkTarget } from "@/core/utilities/common";
+import type { BrowserTargetType } from "@/core/types";
```
Comment (NaMax66) [inline client-app/ui-kit/components/molecules/variations-button/vc-variations-button.vue:29]: remove

File: client-app/ui-kit/components/atoms/product-title/vc-product-title.vue [hunk 5]
```diff
@@ -24,6 +24,7 @@
 
 <script setup lang="ts">
 import { computed } from "vue";
+import type { BrowserTargetType } from "@/core/types";
```
Comment (NaMax66) [inline client-app/ui-kit/components/atoms/product-title/vc-product-title.vue:27]: remove

**Review by NaMax66 — CHANGES_REQUESTED** [review]

**Review by ivan-kalachikov — APPROVED** [review]

**Pull request merged**